contracts/src/v0.8/dev/KeeperRegistryDev.sol

```diff
@@ -542,13 +542,9 @@ contract KeeperRegistryDev is
    * @dev calls the Upkeep target with the performData param passed in by the
    * keeper and the exact gas required by the Upkeep
    */
-  function _performUpkeepWithParams(PerformParams memory params)
-    private
-    nonReentrant
-    validUpkeep(params.id)
-    returns (bool success)
-  {
+  function _performUpkeepWithParams(PerformParams memory params) private nonReentrant returns (bool success) {
     Upkeep memory upkeep = s_upkeep[params.id];
+    if (upkeep.maxValidBlocknumber <= block.number) revert UpkeepCancelled();
```

Comment:
I might be missing something, but shouldn't we have 
```
if (upkeep.maxValidBlocknumber != UINT32_MAX) revert UpkeepCancelled();
```
here? I see this logic was also used before, but we shouldn't allow performs within the cancellation_delay buffer?

Reply:
when an upkeep is cancelled, we set the max valid block number is block.number + cancellation_delay. so within the delay, this if will not be true and the upkeep can be performed. 

if we use upkeep.maxValidBlocknumber != UINT32_MAX, cancellation delay will not have an effect at all?

Reply:
bit of a nitpick, but imho "max valid" should _include_ the terminating value. Because that value is "the maximum that is still valid" - but, alas, this is functionally congruent to what we had before so let's keep it as is. Just wanted to vent. :angel:

Reply:
the purpose of the cancellation buffer is to allow performs (and for node ops to get paid)